tests/test_docker_cat.py

```diff
@@ -0,0 +1,539 @@
+"""Automated integration test of Docker CAT
+
+Run the tests by launching ``pytest`` from the "tests/" folder.
+
+Pytest documentation: https://docs.pytest.org/en/stable/contents.html
+"""
+
+import filecmp
+import os
+import platform
+import re
+import subprocess
+import time
+from pathlib import Path
+
+import docker
+import requests
+
+
+class TestDockerCAT:
+    """This class test the lequal/docker-cat image.
+    It does not build it.
+    Tests can be parametered with environment variables.
+
+    Environment variables:
+        RUN: whether or not to run a container, default "yes", if you
+             already have a running container, set it to "no" and provide
+             information through the other variables.
+        CAT_CONTAINER_NAME: the name to give to the container running
+                            the image.
+        CAT_URL: URL of lequal/docker-cat container if already running
+                 without trailing / from the host.
+                 e.g. http://localhost:9000
+
+    The docstring of each test is the user story it tests.
+    """
+    # Class variables
+    RUN = os.environ.get('RUN', "yes")
+    CAT_CONTAINER_NAME = os.environ.get("CAT_CONTAINER_NAME", "cat")
+    CAT_URL = os.environ.get("CAT_URL", "http://localhost:9000")
+    _DOCKER_CAT_IMAGE = "lequal/docker-cat:latest"
+    _PROJECT_ROOT_DIR = str(Path(os.getcwd()).parent)
+
+    # Functions
+    @classmethod
+    def wait_cat_ready(cls, container_name: str):
+        """This function waits for SonarQube to be configured by
```

Comment:
Codacy found an issue: [1 blank line required between summary line and description (found 0) (D205)](https://app.codacy.com/gh/cnescatlab/docker-cat/pullRequest?prid=6253839)